src/components/chat.tsx

```diff
@@ -357,3 +401,82 @@ export default function Chat(props: ChatProps) {
     </div>
   );
 }
+
+function triggerFeedback(
+  devChatContents: StreamingStepInput[],
+  loading?: boolean,
+): boolean {
+  if (loading) return false;
+  if (devChatContents.length === 0) return false;
+
+  const sinceLastUserMessage = devChatContents.slice(
+    devChatContents.findLastIndex((chat) => chat.role === "user"),
+  );
+
+  if (!sinceLastUserMessage.some((chat) => chat.role === "function"))
+    return false;
+  const lastMessage = sinceLastUserMessage[sinceLastUserMessage.length - 1];
+  const parsed = parseOutput(lastMessage.content);
+
+  return (
+    lastMessage.role === "assistant" &&
+    parsed.tellUser &&
+    parsed.tellUser.trim()[parsed.tellUser.trim().length - 1] !== "?"
+  );
+}
+
+function FeedbackButtons(props: {
+  feedback: "yes" | "no" | null;
+  setFeedback: (feedback: "yes" | "no" | null) => void;
+  isVisible?: boolean;
+  setIsVisible?: (isVisible: boolean) => void;
+}) {
+  const classes = classNames(
+    "sf-flex sf-flex-row",
+    !props.isVisible && "sf-transition-all sf-duration-[1500ms] sf-opacity-0",
+  );
+
+  return (
+    <div className={classes}>
+      <div className="sf-flex sf-flex-col sf-place-items-center sf-gap-y-1 sm:sf-text-sm sf-text-md">
+        <div className="sf-flex sf-flex-row sf-gap-x-4 sf-px-2 sf-whitespace-nowrap ">
+          Did this answer your question?
+        </div>
+        <div className="sf-flex sf-flex-row sf-gap-x-2">
+          <button
```

Comment:
Personally I think that we should give a textbox here when `no` has been clicked, asking the user why

Then the user can tell it why. This will be really useful for debugging/understanding what issues are :smile: